### PR TITLE
Rework genproj to use gensources to build sources list for each profile and host platform

### LINKED ISSUE #8046: Genproj includes only Linux platform C# specific files
We need to include platform-specific files for example `external/corefx/src/Common/src/Interop/OSX/Interop.Libraries.cs` from `darwin_net_4_x_System.Data.dll.sources` to build correctly on non-Linux platforms

### PR DESCRIPTION
**Pull request by kg**
Fixes #8046

### CODE REVIEW

File: mcs/build/executable.make [hunk 1]
```diff
@@ -192,6 +192,7 @@ csproj-local:
 	(echo $(is_boot); \
 	echo $(USE_MCS_FLAGS) $(LIBRARY_FLAGS) $(LIB_MCS_FLAGS) $(patsubst %,-r:%,$(LIB_REFS)); \
 	echo $(PROGRAM); \
```
Comment (luhenry) [inline mcs/build/executable.make]: Do we need to duplicate it?

File: msvc/scripts/Makefile [hunk 2]
```diff
@@ -1,8 +1,8 @@
 all: genproj.exe prepare.exe
 	mono --debug genproj.exe
 
-genproj.exe: genproj.cs
-	mcs -debug:portable -g genproj.cs -r:System.Xml.Linq
+genproj.exe: genproj.cs ../../mcs/build/gensources.cs
+	mcs -debug:portable genproj.cs ../../mcs/build/gensources.cs -main:Driver -r:System.Xml.Linq -r:System.dll -r:System.Core.dll
```
Comment (luhenry) [inline msvc/scripts/Makefile]: Should we use csc instead?

**Comment by kg** [comment]
@monojenkins build Linux x64 FullAOT+LLVM

**Comment by kg** [comment]
The one other big problem I forgot to mention is this:
screenshot 2018-06-28 10 48 40
I think it's a VS bug as I only started seeing it after an update. It goes away if you Reload jay, and I haven't touched the vcxproj, so it's an annoyance at most. I hope I can fix it in a follow-up PR somehow though.

**Comment by marek-safar** [comment]
@monojenkins commit csproj

**Comment by akoeplinger** [comment]
Awesome work, thank you @kg !

**Pull request merged**